pkg/util/concurrent_map.go

```diff
@@ -0,0 +1,319 @@
+package util
+
+import (
+	"encoding/json"
+	"sync"
+)
+
+/*
+	This module is modified from github.com/orcaman/concurrent-map.
+	MIT license.
+ */
+
+var SHARD_COUNT = 32
```

Comment:
can we use a config instead of const 32?

I worry sometimes 32 shardings may also have a high contention if our test concurrency is high.

Reply:
Sure